DependencyInjection/Configuration.php

```diff
@@ -156,7 +158,13 @@ private function getDbalConnectionsNode(): ArrayNodeDefinition
             ->fixXmlConfig('default_table_option')
             ->children()
                 ->scalarNode('driver')->defaultValue('pdo_mysql')->end()
-                ->scalarNode('platform_service')->end()
+                ->scalarNode('platform_service')
+                    ->setDeprecated(
+                        'doctrine/doctrine-bundle',
+                        '2.9',
+                        'The "platform_service" configuration key is deprecated since doctrine-bundle 2.9. DBAL 4 will not support setting a custom platform via connection params anymore.'
```

Comment:
Hello, what is the new way to change the platform ?

Reply:
drivers are responsible for providing the platform. So using a driver middleware (the main extension point of DBAL) is probably the way to go for your case.

Reply:
I think you can achieve this with a custom DBAL driver middleware